### DISCUSSION TITLE
Function as question parameter

### DISCUSSION THREAD

**Discussion by danpilat** [jspsych/jsPsych#599] [Q&A]
Hi there, 

I have two trials, one following each other. I want to use the responses collected in the first trial, to show them in the prompt in the trial that follows. I know that the prompt needs to be wrapped into a function() in order for JSpsych to re-evaluate it, otherwise the value comes up as undefined. However, I do not know how to wrap the question or the prompt in a function. 

This is where I capture the value the first time:

```js
var task_how_q1_block ={
type: 'survey-text',
questions: [{prompt: "#1...."}],
on_finish: function(data) {
   var responses = JSON.parse(data.responses);
    topsavingway = responses.Q0;
    jsPsych.data.addProperties({subject:subject_id, topWay: topsavingway});
}
};
```
And this is where I want to use it:

```js
var task_how_elab_block ={
type: 'survey-text',
questions: [{prompt: "Please elaborate on how" + jsPsych.data.get().select('topWay').values[0] + "is a good way to do stuff. (10 words minimum).", rows: 5, columns: 40}]
};
```

However, using prompt: function() { return....} does not work. 

Thanks!
Dan

**Comment by jodeleeuw** [comment]
Hi Dan,

You can wrap your `questions` parameter in a function:

```js
var task_how_elab_block ={
  type: 'survey-text',
  questions: function(){
    return [
      {prompt: "Please elaborate on how" + jsPsych.data.get().select('topWay').values[0] + "is a good way to do stuff. (10 words minimum).", rows: 5, columns: 40}
    ]
  }
};
```

It would be nice to support wrapping individual pieces of parameters (like just `prompt` in the `questions`) as functions, but that isn't supported yet. That might be a nice addition to a future version!
